### PR TITLE
Document how to profile Ray using pprof.

### PR DESCRIPTION
**Pull request by concretevitamin**
## What do these changes do?
Add documentation.

## Related issue number
N/A.

### CODE REVIEW

**Comment by pcmoritz** [comment]
LGTM, let's also link it from the main documentation

**Comment by richardliaw** [comment]
@concretevitamin there's an linting error that's getting thrown

**Comment by concretevitamin** [comment]
@richardliaw rst is hard!

@pcmoritz done

**Comment by robertnishihara** [comment]
Thanks for posting this! Trying it out now :)

File: doc/source/profiling.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Profiling Ray
```
Comment (robertnishihara) [inline doc/source/profiling.rst]: Does this work? I thought rst requires the underlining to be the same length as the title.

**Comment by robertnishihara** [comment]
This is super cool. Can you accept this PR? https://github.com/concretevitamin/ray/pull/1

I made some changes to make the instructions easier to follow (at least for me).

**Review by robertnishihara — APPROVED** [review]
Let's just make sure the linting passes on Travis and then merge it.

**Review by elibol — APPROVED** [review]
Thanks for putting this together! I gave it a try. To run `plasma_manager` with profiling, I did this:
```
env = os.environ.copy()
env["LD_PRELOAD"] = "/usr/lib/libprofiler.so"
env["CPUPROFILE"] = "/tmp/pprof.out"
process = subprocess.Popen(command,
env=env,
stdout=stdout_file,
stderr=stderr_file)
```

pprof

**Comment by pcmoritz** [comment]
Thanks everybody :)

**Pull request merged**